rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -1029,4 +1005,30 @@ void forget() {
             }
         }
     }
+
+    @Override
+    public boolean running() {
+        return status.equals(LRAStatus.Active);
+    }
+
+    @Override
+    public int cancel() {
+        abortLRA();
+
+        if (status.equals(LRAStatus.Cancelled)) {
```

Comment:
Just a question, is it correct to verify the LRA status only against the `Cancelled`? It seems that other LRA statuses could not be properly converted to `ActionStatus` codes.

Reply:
If the LRA is not in the `Cancelled` state we return the ArjunaCore ActionStatus to the reaper (it is on the next line 2022):

```
return status();
```